pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -156,6 +156,13 @@ func updateVitessShard(key client.ObjectKey, vts *planetscalev2.VitessShard, vtk
 	// Update labels, but ignore existing ones we don't set.
 	update.Labels(&vts.Labels, newShard.Labels)
 
+	updateVitessShardAnnotations(vts, newShard)
+
+	// For now, everything in Spec is safe to update.
+	vts.Spec = newShard.Spec
+}
+
+func updateVitessShardAnnotations(vts *planetscalev2.VitessShard, newShard *planetscalev2.VitessShard) {
```

Comment:
Nit: move this below `updateVitessShardInPlace` since it's a helper function.

Reply:
Good idea.

Reply:
Done.